### ISSUE TITLE
lorotate enhancement

### ISSUE THREAD

**Issue by JTakkalaki** [logrotate/logrotate#577]
Presently, the compression process involves initially copying the source log file to the destination file before initiating compression. This method poses concerns, especially on SSD devices, where write disk operations are critical. An improved approach would involve implementing compression without the necessity of copying the entire source log file to the destination file. Instead, compression could be performed directly on the destination file once the threshold limit is reached, reading from the source file as needed. Then truncate the source file portion, which has already been compressed.

**Comment by kdudka** [comment]
In the default configuration, logrotate changes only name of the log file.  logrotate does not copy the content of the log.

**Comment by JTakkalaki** [comment]
@kdudka ,

I'm not referring to the default configuration; this relates to compression when reaching the threshold for the log limit.
The configuration is as follows:
- compress
- dateext
- dateformat -%Y%m%d_%H%M%S
- missingok

**Comment by kdudka** [comment]
I do not think that any of the above directives would cause log content to be copied on the file system before compression.  Of course, `copytruncate` would be another story...
